packages/clay-table/src/index.tsx

```diff
@@ -0,0 +1,126 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import classNames from 'classnames';
+
+import Body, {TableBodyType} from './Body';
+import Cell, {TableCellType} from './Cell';
+import Head, {TableHeadType} from './Head';
+import Row, {TableRowType} from './Row';
+
+interface Props extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * This property vertically align the contents
+	 * inside the table body according a given position.
+	 */
+	bodyVerticalAlignment?: VerticalAlignmentType;
+
+	/**
+	 * Applies a Bordered style on Table.
+	 */
+	bordered?: boolean;
+
+	/**
+	 * This property keeps all the headings on one line.
+	 */
+	headingNoWrap?: boolean;
+
+	/**
+	 * This property vertically align the contents
+	 * inside the table header according a given position.
+	 */
+	headVerticalAlignment?: VerticalAlignmentType;
+
+	/**
+	 * Applies a Hover style on Table.
+	 */
+	hover?: boolean;
+
+	/**
+	 * This property enables keeping everything on one line.
+	 */
+	noWrap?: boolean;
+
+	/**
+	 * Turns the table responsive.
+	 */
+	responsive?: boolean;
+
+	/**
+	 * Defines the responsive sizing.
+	 */
+	responsiveSize?: ResposiveSizeType;
+
+	/**
+	 * Applies a Striped style on Table.
+	 */
+	striped?: boolean;
+
+	/**
+	 * This property vertically align the contents
+	 * inside the table according a given position.
+	 */
+	tableVerticalAlignment?: VerticalAlignmentType;
+}
+
+type ResposiveSizeType = 'lg' | 'md' | 'sm' | 'xl';
+
+type VerticalAlignmentType = 'bottom' | 'middle' | 'top';
+
+const ClayTable: React.FunctionComponent<Props> & {
+	Body: TableBodyType;
+	Cell: TableCellType;
+	Head: TableHeadType;
+	Row: TableRowType;
+} = ({
+	bodyVerticalAlignment,
+	bordered,
+	children,
+	className,
+	headVerticalAlignment,
+	headingNoWrap,
+	hover,
+	noWrap,
+	responsive,
+	responsiveSize,
+	striped,
+	tableVerticalAlignment,
+	...otherProps
+}) => {
+	return (
+		<div
```

Comment:
I'm not sure but the table does not always need a wrap.

Reply:
`.table-responsive and .table-responsive-{sm|md|lg|xl} should be placed in a div that wraps the table element. See https://github.com/twbs/bootstrap/issues/24638 for more info.
`

Reply:
@pat270 apparently this issue was fixed by https://github.com/twbs/bootstrap/pull/25058. Would be a good idea remove this wrap?

Reply:
https://getbootstrap.com/docs/4.0/content/tables/#responsive-tables

Reply:
.div wrapper removed!